src/SmartComponents/BaselinesPage/EditBaseline/FactModal/FactModal.js

```diff
@@ -0,0 +1,243 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Button, Checkbox, Modal, TextInput } from '@patternfly/react-core';
+
+import { factModalActions } from '../FactModal/redux';
+import { baselinesTableActions } from '../../../BaselinesTable/redux';
+import editBaselineHelpers from '../helpers';
+
+class FactModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.cancelFact = this.cancelFact.bind(this);
+        this.confirmModal = this.confirmModal.bind(this);
+        this.renderCategoryCheckbox = this.renderCategoryCheckbox.bind(this);
+        this.renderFactInput = this.renderFactInput.bind(this);
+        this.renderValueInput = this.renderValueInput.bind(this);
+        this.renderModalBody = this.renderModalBody.bind(this);
+
+        this.state = {
+            factName: this.props.factName,
+            valueName: this.props.valueName,
+            factData: this.props.factData
+        };
+
+        if (this.props.factName !== '' && this.props.valueName === '') {
+            this.state.categoryCheck = true;
+        } else {
+            this.state.categoryCheck = false;
+        }
+
+        this.handleChange = (checked) => {
+            this.setState({ categoryCheck: checked });
+        };
+
+        this.handleNewName = value => {
+            this.setState({ factName: value });
+        };
+
+        this.handleNewValue = value => {
+            this.setState({ valueName: value });
+        };
+    }
+
+    cancelFact() {
+        const { toggleFactModal } = this.props;
+
+        toggleFactModal();
+    }
+
+    confirmModal() {
+        const { toggleFactModal, factName, valueName } = this.props;
+
+        if ((factName === '' && valueName === '')
+        || (factName === undefined && valueName === undefined)) {
+            this.addFact();
+        } else {
+            this.editFact();
+        }
+
+        toggleFactModal();
+    }
+
+    addFact() {
+        const { categoryCheck, factName, valueName, factData } = this.state;
+        const { baselineData, patchBaseline } = this.props;
+
+        let newFactData = editBaselineHelpers.buildNewFactData(categoryCheck, factName, valueName, factData);
+        let newAPIBody = editBaselineHelpers.makeAddFactPatch(newFactData, baselineData, factData);
+        patchBaseline(baselineData.id, newAPIBody);
+    }
+
+    editFact() {
+        const { categoryCheck, factName, valueName, factData } = this.state;
+        const { baselineData, patchBaseline } = this.props;
+
+        let editedFactData = editBaselineHelpers.buildEditedFactData(
+            categoryCheck, this.props.factName, factName, this.props.valueName, valueName, factData
+        );
```

Comment:
This function and `addFact` look to be the same except for this one line. Any DRY we could apply here? A `saveFact` function perhaps that does one thing if its an add and another if its a delete?